nuxeo-aggregation/nuxeo-aggregation-navigation.html

```diff
@@ -0,0 +1,232 @@
+        height: 2px;
+        background: rgba(0, 0, 0, 0.15);
+        pointer-events: none;
+      }
+
+      #cursor .label {
+        position: absolute;
+        right: 56px;
+        top: -14px;
+        white-space: nowrap;
+        text-align: center;
+        background: black;
+        color: white;
+        padding: 6px 8px;
+        font-size: 0.8rem;
+        font-weight: bold;
+        border-radius: 4px;
+        min-width: 64px;
+      }
+    </style>
+
+    <div id="keys" on-mouseout="_mouseOut" on-tap="_tap">
+      <template is="dom-repeat" items="[[_keys]]" as="key">
+        <div class="key" on-mousemove="_mouseMove"
+             style$="top: [[key.top]]px; height: [[key.height]]px; color: [[_color(key.visible)]];">
+          [[_label(key)]]
+        </div>
+      </template>
+      <div id="cursor">
+        <div class="label">[[_cursorLabel]]</div>
+      </div>
+    </div>
+
+  </template>
+
+  <script>
+    Polymer({
+      is: 'nuxeo-aggregation-navigation',
+      behaviors: [Nuxeo.I18nBehavior],
+      properties: {
+
+        buckets: {
+          type: Array,
+          value: [],
+          observer: '_bucketsChanged'
+        },
+
+        granularity: {
+          type: Number,
+          value: 30
+        },
+
+        opacity: {
+          type: Number,
+          value: 0.85
+        },
+
+        _keys: {
+          type: Array,
+          value: []
+        },
+
+        _count: {
+          type: Number,
+          value: 0
+        },
+
+        _cursorIndex: {
+          type: Number,
+          value: 0
+        },
+
+        _cursorLabel: {
+          type: String,
+          value: ''
+        },
+
+        _rect: {
+          type: Object,
+          value: {top: 0, right: 0, left: 0, bottom: 0}
+        }
+      },
+
+      attached: function() {
+        this.addEventListener('mouseover', function() {
+          this._visibility(true);
+        }.bind(this));
+
+        this.addEventListener('mouseout', function() {
+          this._visibility(false);
+        }.bind(this));
+      },
+
+      _visibility: function(visible) {
+        this.$.keys.style.visibility = visible ? 'visible' : 'hidden';
+        this.style.background = visible ? 'rgba(255, 255, 255, 0.85)' : 'transparent';
+      },
+
+      _label: function(key) {
+        return this.i18n(key.name);
+      },
+
+      _bucketsChanged: function(buckets) {
+        this._rect = this.$.keys.getBoundingClientRect();
+
+        // count documents in buckets
+        this._count = 0;
+        buckets.forEach(function(bucket) {
+          this._count += bucket.docCount;
+        }.bind(this));
+        this.style.opacity = this._count ? this.opacity : 0;
+
+        // fill keys array based on buckets
+        this.set('_keys', []);
+        var offset = 1;
+        var granularity = this.granularity;
+        buckets.forEach(function(bucket) {
+
+          var visible = false;
+          if (granularity >= this.granularity) {
```

Comment:
not sure I understand what is `granularity` since it is defined as `this.granularity` line 184

Reply:
In line 184 local `granularity` var is initialized based on `this.granularity` than when iterating in buckets it will be changed accordingly. Granularity purpose is to have a way to control the distance between buckets' labels and avoid labels overlapping.